### PR TITLE
[MIG]Migrated Business Requirement Deliverable Resource Template from 8.0 to 10.0.

### PR DESCRIPTION
**Pull request by sudhir-serpentcs**
Issue #48 
- This PR is depend on #224 
- Migrated Business Requirement Deliverable from v8 to v10.

### CODE REVIEW

**Review by elicoidal — APPROVED** [review]
LGTM

**Comment by elicoidal** [comment]
@seb-elico @victormartinelicocorp

**Comment by martintorresvm** [comment]
@elicoidal Need #223 #224 merged and this re-base as soon is done. Any later issue could not be discovered now.

File: business_requirement_deliverable_resource_template/tests/test_br_resource_template.py
Comment (seb-elico) [inline business_requirement_deliverable_resource_template/tests/test_br_resource_template.py:27]: Out of curiosity, why changing the product?

Comment (sudhir-serpentcs) [reply business_requirement_deliverable_resource_template/tests/test_br_resource_template.py:27]: @seb-elico The product wasn't exist in v10 product module. So we changed it by other service type of product.

**Review by seb-elico — APPROVED** [review]
LGTM

**Review by martintorresvm — CHANGES_REQUESTED** [review]
@sudhir-serpentcs @elicoidal  @seb-elico 
UT was designed before just to pass coveralls. 

Need to improve quality of Test.

Extend the BRD test so the DL product change should be covered by the tested before this module.

Now specific here is the fact we need test whether the DL have generated the RL with same information as the product template resources.

For instance: 
def test_product_id_onchnage(self):
(by the way name is wrong)
def test_product_id_onchange(self):
A)Setup the condition to test. (done in def setUp)
B)Execute the function. (product_id_change)
C)Check with assert the value or values expected.

**Comment by seb-elico** [comment]
Agreed with @victormartinelicocorp

**Comment by elicoidal** [comment]
@sudhir-serpentcs Let's try to finish this today :smile:

**Comment by elicoidal** [comment]
What is the situation here?

**Comment by elicoidal** [comment]
@victormartinelicocorp Can you review the tests so that we can move forward?

**Comment by martintorresvm** [comment]
@sudhir-serpentcs Travis first error is on here:

======== Testing test_flake8 ========

business_requirement_deliverable_resource_template/tests/test_br_deliverable.py:5:1: F401 'odoo.exceptions.ValidationError' imported but unused

@elicoidal  Others errors cause dependency on BRD. So need to rebase this PR with #224 once is merged

**Comment by elicoidal** [comment]
#224 and #223 are merged: please rebase

**Comment by elicoidal** [comment]
@victormartinelicocorp can you recheck your review?

**Review by martintorresvm — CHANGES_REQUESTED** [review]
@YogeshMahera-SerpentCS 

Please move all `_demo.xml` files into folder /demo as OCA guidelines suggest.

https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#directories

**Comment by elicoidal** [comment]
Functional tests are OK

**Comment by elicoidal** [comment]
@victormartinelicocorp

**Comment by martintorresvm** [comment]
> @YogeshMahera-SerpentCS
Please move all _demo.xml files into folder /demo as OCA guidelines suggest.
https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#directories

@sudhir-serpentcs @YogeshMahera-SerpentCS Could have a check? 
Is just move demo files to the right folder, if too busy let me know.

cc @elicoidal

**Comment by victormartinelicocorp** [comment]
> @YogeshMahera-SerpentCS
Please move all _demo.xml files into folder /demo as OCA guidelines suggest.
https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#directories

@sudhir-serpentcs @YogeshMahera-SerpentCS Could have a check? 
Is just move demo files to the right folder, if too busy let me know.

cc @elicoidal

**Comment by YogeshMahera-SerpentCS** [comment]
@victormartinelicocorp 
Please review

**Review by martintorresvm — APPROVED** [review]

**Pull request merged**
